firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/BookmarksSaver.swift

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import MozillaAppServices
+import Shared
+
+protocol BookmarksSaver {
+    /// Saves or updates a bookmark
+    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<Void, Error>
+}
+
+struct DefaultBookmarksSaver: BookmarksSaver {
+    enum SaveError: Error {
+        case bookmarkTypeDontSupportSaving
+    }
+
+    let profile: Profile
+
+    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<Void, Error> {
+        return await withCheckedContinuation { continuation in
+            let operation: Success? = {
+                switch bookmark.type {
+                case .bookmark:
+                    // no parent create it
```

Comment:
I'm not sure what this comment means, does it mean the bookmark doesn't have a folder parent?

Reply:
Yes. I removed it since it can be also understand by the code.